packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -835,7 +835,14 @@ class _DraggableScrollableSheetScrollController extends ScrollController {
         curve: Curves.linear,
       );
     }
-    extent.updateSize(extent.initialSize, position.context.notificationContext!);
+    final StatefulElement element = position.context.notificationContext! as StatefulElement;
```

Comment:
This is very different from the earlier fix, can you explain a bit in the PR description what you found as the root issue, and why this fixes it?

Reply:
i added description

Reply:
Hmm, ok, thank you for adding the comment.
This does not look like the right fix to me.

We should not need to be checking whether or not the element is dirty, and further shouldn't be using a post frame callback to initiate the update. It feels like this is working around the root cause of the issue, do you know why we end up here with a dirty element? Tracing this back further will probably reveal the real issue.